lib/parse_json.c

```diff
@@ -82,6 +82,7 @@ static json_object *getobj(json_object *jo, const char *key1, const char *key2)
         assert(key2 == NULL);
         return jo;
     } else if (!json_object_is_type(jo, json_type_object)) {
+        warnx("json_object_is_type is %s for %s", json_type_to_name(json_object_get_type(jo)), key1);
```

Comment:
So unfortunately, this causes test_cfg to fail because it's comparing parser output, and this branch gets hit.

This is due to a design choice that json-c made around NULL objects.  For a `json_object *jo`, the fact that `jo == NULL` means that it's unitialized.  However, if `json_object_get_type(jo) == json_type_null`, that means that jo is equivalent to "not found in file", or was specified as literal null.  (These are basically equivalent.)

Adding to this confusion: the type of a `json_object` may or may not be `json_type_object`.  Our parser interface hides this weirdness and instead only talks about "dicts" and "arrays" in the hope of not being ambiguous.

In the YAML parser, there's a rudimentary trace system - one can `#define DEBUG 1` and get more information about what's happening.  This could be expanded to the other parsers if desired: I didn't do that only because it wasn't needed to make them work.  If getting this output is required without rebuilding rpminspect with different flags, it might also work to use an environment variable to turn on debug printing - like $KRB5_TRACE does.

Reply:
Since the json parser currently has the luxury of not needing to support backward compat, another possibility is to modify this to only print the warning when the type also isn't null.  That would only catch type errors.